plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/linked-product-control.tsx

```diff
@@ -115,50 +135,116 @@ const LinkedProductControl = ( {
 	location: WooCommerceBlockLocation;
 	usesReference: string[] | undefined;
 } ) => {
-	const [ isDropdownOpen, setIsDropdownOpen ] = useState< boolean >( false );
-	const { product, isLoading } = useGetProduct( query.productReference );
+	const referenceType = 'product';
+	const isProductReferenceAvailable = location.type === referenceType;
+	const { productReference } = query;
 
+	const { product, isLoading } = useGetProduct( productReference );
+	const [ isDropdownOpen, setIsDropdownOpen ] = useState< boolean >( false );
+	const [ radioControlState, setRadioControlState ] = useState< PRODUCT_REF >(
+		isProductReferenceAvailable && isEmpty( productReference )
+			? PRODUCT_REF.CURRENT_PRODUCT
+			: PRODUCT_REF.SPECIFIC_PRODUCT
+	);
+	const prevReference = useRef< number | undefined >( undefined );
 	const showLinkedProductControl = useMemo( () => {
-		const isInRequiredLocation = usesReference?.includes( location.type );
-		const isProductContextRequired = usesReference?.includes( 'product' );
-		const isProductContextSelected =
-			( query?.productReference ?? null ) !== null;
-
-		return (
-			isProductContextRequired &&
-			! isInRequiredLocation &&
-			isProductContextSelected
-		);
-	}, [ location.type, query?.productReference, usesReference ] );
+		const isProductContextRequired =
+			usesReference?.includes( referenceType );
+
+		return isProductContextRequired;
+	}, [ usesReference ] );
 
 	if ( ! showLinkedProductControl ) return null;
 
+	const showRadioControl = isProductReferenceAvailable;
+	const showDropdown = showRadioControl
+		? radioControlState === PRODUCT_REF.SPECIFIC_PRODUCT
+		: ! isEmpty( productReference );
+
+	const radioControlHelp =
+		radioControlState === PRODUCT_REF.CURRENT_PRODUCT
+			? __(
+					'Related products will be pulled from the product a shopper is currently viewing',
+					'woocommerce'
+			  )
+			: __(
+					'Select a product to pull the related products from',
+					'woocommerce'
+			  );
+
```

Comment:
Agree! I changed to:

- 'Linked products will be pulled from the product a shopper is currently viewing'
- 'Select a product to pull the linked products from'

In future iterations, we can make it more contextual. Or actually, it could part of collections PR (Upsells, Related Products) to modify these. Since the collections aren't yet merged I don't want to include that in this PR 🙌

Reply:
Sounds good to me. Thanks for making changes 🚀